src/business-logic/services/test/rundown-timeline-service.spec.ts

```diff
@@ -77,6 +82,97 @@ describe(RundownTimelineService.name, () => {
       await expect(result).rejects.toThrow(AlreadyActivatedException)
     })
   })
+
+  describe(`${RundownTimelineService.prototype.activateRundown.name}`, () => {
+    beforeEach(() => jest.useFakeTimers())
+    afterEach(() => jest.useRealTimers())
+    it('does not emit events unless piecess are changed on activateRundown', async () => {
+      const aRundown: Rundown = EntityMockFactory.createRundown({ id: 'aRundown', isRundownActive: false })
+      const rundowns: Rundown[] = [aRundown]
+      const mockRundownRepository: RundownRepository = mock<RundownRepository>()
+      when(mockRundownRepository.getRundown(aRundown.id)).thenResolve(aRundown)
+      when(mockRundownRepository.getBasicRundowns()).thenResolve(rundowns)
+      const mockRundownEventEmitter: RundownEventEmitter = mock<RundownEventEmitter>()
+
+      const testee: RundownTimelineService = createTestee({
+        rundownRepository: instance(mockRundownRepository),
+        rundownEventEmitter: instance(mockRundownEventEmitter),
+      })
+
+      await testee.activateRundown('aRundown')
+      verify(mockRundownEventEmitter.emitInfinitePiecesUpdatedEvent(aRundown)).once()
+      verify(mockRundownEventEmitter.emitActivateEvent(aRundown)).once()
+      verify(mockRundownEventEmitter.emitSetNextEvent(aRundown)).once()
+    })
+  })
+
+  describe(`${RundownTimelineService.prototype.takeNext.name}`, () => {
+    it('does not emit events unless piecess are changed on takeNext', async () => {
+      const activePiece: Piece = EntityTestFactory.createPiece({ id: 'activePiece' })
```

Comment:
Your test is passing, but the implementation in `RundownTimelineService` is checking on the wrong conditional (mentioned in a previous comment). 
The reason your test is still parsing is because a bug (that this test just discovered for us) in `EntityTestFactory.createPiece()`. `createPiece()` does currently not set a default `PieceLifespan` for the Piece. So unless you provide a `PieceLifespan` it will be undefined.
This result in the Rundown thinking that the Piece is an infinite Piece because the Rundown checks `piece.pieceLifespan !== PieceLifspan.WITHIH_PART`, which is true then the lifespan is undefined.

To fix the bug: `pieceLifespan: PieceLifespan.WITHIN_PART` should be added to the `createPiece` method in `EntityTestFactory` so it would look like this:
![image](https://github.com/tv2/sofie-server/assets/23049647/3437b2ac-7689-45c0-b2ad-9bcc833bed45)

Adding this to `createPiece` will cause your test to fail because now there will be an emitted `infinitePiecesUpdatedEvent` since the current implementation emits when the InfinitePieces array is the same before and after the operation which they will be now, since there is no infinitePieces in the Rundown before activation and there is no infinite Pieces in the Rundown after activation.